src/transaction/TransactionController.test.ts

```diff
@@ -666,30 +661,25 @@ describe('TransactionController', () => {
   });
 
   it('should fail to approve an invalid transaction', async () => {
-    await new Promise(async (resolve) => {
-      const controller = new TransactionController({
-        provider: PROVIDER,
-        sign: () => {
-          throw new Error('foo');
-        },
-      });
-      controller.context = {
-        NetworkController: MOCK_NETWORK,
-      } as any;
-      controller.onComposed();
-      const from = '0xe6509775f3f3614576c0d83f8647752f87cd6659';
-      const to = '0xc38bf1ad06ef69f0c04e29dbeb4152b4175f0a8d';
-      const { result } = await controller.addTransaction({ from, to });
-      result.catch((error) => {
-        const { transaction, status } = controller.state.transactions[0];
-        expect(transaction.from).toBe(from);
-        expect(transaction.to).toBe(to);
-        expect(status).toBe(TransactionStatus.failed);
-        expect(error.message).toContain('foo');
-        resolve('');
-      });
-      await controller.approveTransaction(controller.state.transactions[0].id);
+    const controller = new TransactionController({
+      provider: PROVIDER,
+      sign: () => {
+        throw new Error('foo');
+      },
     });
+    controller.context = {
+      NetworkController: MOCK_NETWORK,
+    } as any;
+    controller.onComposed();
+    const from = '0xe6509775f3f3614576c0d83f8647752f87cd6659';
+    const to = '0xc38bf1ad06ef69f0c04e29dbeb4152b4175f0a8d';
+    const { result } = await controller.addTransaction({ from, to });
+    const { transaction, status } = controller.state.transactions[0];
+    expect(transaction.from).toBe(from);
+    expect(transaction.to).toBe(to);
+    expect(status).toBe(TransactionStatus.unapproved);
+    await controller.approveTransaction(controller.state.transactions[0].id);
+    await expect(result).rejects.toThrow('foo');
```

Comment:
There was a change in ordering here - the `controller.state.transactions[0];` checks should happen _after_ the transaction is rejected. That's why the status went from failed to unapproved here.

Reply:
Similar issue with some of the other tests in this module

Reply:
🤦‍♀️